Zope 4.6.2 compared to 4.5.5 updates `zope.component` from 4.6.2 to 5.0.0.
The 2.7 tests completely fail because Products.Archetypes is not compatible with this. Should be fixed with https://github.com/plone/Products.Archetypes/pull/137.

But the Python3 tests fail as well, [for example](https://jenkins.plone.org/job/pull-request-5.2-3.6/1938/testReport/junit/plone.app.users/tests/password_rst/):

```
File "/home/jenkins/workspace/pull-request-5.2-3.6/src/plone.app.users/plone/app/users/tests/password.rst", line 27, in password.rst
Failed example:
    browser.open('http://nohost/plone/' + view_name)
Expected:
    Traceback (most recent call last):
    ...
    zExceptions.unauthorized.Unauthorized: ...You are not authorized to access this resource...
Got:
    Traceback (most recent call last):
      File "/srv/python3.6/lib/python3.6/doctest.py", line 1330, in __run
        compileflags, 1), test.globs)
      File "<doctest password.rst[8]>", line 1, in <module>
        browser.open('http://nohost/plone/' + view_name)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.testbrowser-5.5.1-py3.6.egg/zope/testbrowser/browser.py", line 256, in open
        self._processRequest(url, make_request)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.testbrowser-5.5.1-py3.6.egg/zope/testbrowser/browser.py", line 282, in _processRequest
        resp = make_request(reqargs)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.testbrowser-5.5.1-py3.6.egg/zope/testbrowser/browser.py", line 253, in make_request
        return self.testapp.get(url, **args)
      File "/home/jenkins/.buildout/eggs/cp36m/WebTest-2.0.35-py3.6.egg/webtest/app.py", line 328, in get
        expect_errors=expect_errors)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.testbrowser-5.5.1-py3.6.egg/zope/testbrowser/browser.py", line 94, in do_request
        expect_errors)
      File "/home/jenkins/.buildout/eggs/cp36m/WebTest-2.0.35-py3.6.egg/webtest/app.py", line 623, in do_request
        res = req.get_response(app, catch_exc_info=True)
      File "/home/jenkins/.buildout/eggs/cp36m/WebOb-1.8.7-py3.6.egg/webob/request.py", line 1310, in send
        application, catch_exc_info=True)
      File "/home/jenkins/.buildout/eggs/cp36m/WebOb-1.8.7-py3.6.egg/webob/request.py", line 1278, in call_application
        app_iter = application(self.environ, start_response)
      File "/home/jenkins/.buildout/eggs/cp36m/WebTest-2.0.35-py3.6.egg/webtest/lint.py", line 201, in lint_app
        iterator = application(environ, start_response_wrapper)
      File "/home/jenkins/.buildout/eggs/cp36m/plone.testing-8.0.3-py3.6.egg/plone/testing/_z2_testbrowser.py", line 39, in wrapped_func
        return func(*args, **kw)
      File "/home/jenkins/.buildout/eggs/cp36m/plone.testing-8.0.3-py3.6.egg/plone/testing/_z2_testbrowser.py", line 66, in __call__
        wsgi_result = publish(environ, start_response)
      File "/home/jenkins/.buildout/eggs/cp36m/Zope-4.6.2-py3.6.egg/ZPublisher/WSGIPublisher.py", line 385, in publish_module
        request.close()
      File "/home/jenkins/.buildout/eggs/cp36m/Zope-4.6.2-py3.6.egg/ZPublisher/BaseRequest.py", line 216, in close
        notify(EndRequestEvent(None, self))
      File "/home/jenkins/.buildout/eggs/cp36m/zope.event-4.5.0-py3.6.egg/zope/event/__init__.py", line 32, in notify
        subscriber(event)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.component-5.0.0-py3.6.egg/zope/component/event.py", line 27, in dispatch
        component_subscribers(event, None)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.component-5.0.0-py3.6.egg/zope/component/_api.py", line 134, in subscribers
        return sitemanager.subscribers(objects, interface)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.interface-5.4.0-py3.6-linux-x86_64.egg/zope/interface/registry.py", line 448, in subscribers
        return self.adapters.subscribers(objects, provided)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.interface-5.4.0-py3.6-linux-x86_64.egg/zope/interface/adapter.py", line 895, in subscribers
        subscriptions = self.subscriptions([providedBy(o) for o in objects], provided)
      File "/home/jenkins/.buildout/eggs/cp36m/zope.interface-5.4.0-py3.6-linux-x86_64.egg/zope/interface/adapter.py", line 877, in _uncached_subscriptions
        if order >= len(byorder):
      File "/home/jenkins/shiningpanda/jobs/8809464e/virtualenvs/0dd3bfc1/lib/python3.6/collections/__init__.py", line 1049, in __len__
        def __len__(self): return len(self.data)
      File "/home/jenkins/.buildout/eggs/cp36m/ZODB-5.6.0-py3.6.egg/ZODB/Connection.py", line 785, in setstate
        raise ConnectionStateError(msg)
    ZODB.POSException.ConnectionStateError: Shouldn't load state for persistent.list.PersistentList 0x19cb2f31a941580b when the connection is closed
```

This is when visiting `@@change-password` as anonymous user. When I try it manually, it works: you get redirected to the login form as expected. But a `ConnectionStateError` in the tests does not sounds good.
There are [8 failures](https://jenkins.plone.org/job/pull-request-5.2-3.6/1938/), some of them similar to the above.

Maybe they can be fixed. But it makes we wonder if we should stick to the previous `zope.component` 4.6.2.
And maybe stick to `zope.interface` 4.2.0 as well, instead of upgrading to 4.4.0.
(For Plone 6: sure, use the latest.)

Some possible causes of problems are these two issues/PRs:
- https://github.com/zopefoundation/zope.component/issues/51
- https://github.com/zopefoundation/zope.interface/pull/228

--- a/versions.cfg
+++ b/versions.cfg
@@ -2,7 +2,7 @@
 # This file is part of buildout.coredev repository: https://github.com/plone/buildout.coredev
 # The plone release team is responsible for it,
 # if you have suggestions, please open an issue at: https://github.com/plone/buildout.coredev/issues
-extends = https://zopefoundation.github.io/Zope/releases/4.5.5/versions.cfg
+extends = https://zopefoundation.github.io/Zope/releases/4.6.2/versions.cfg
 
 [versions]
 ##############################################################################
@@ -83,7 +83,7 @@ zope.keyreference = 4.2.0
 zope.mkzeoinstance = 4.1
 zope.password = 4.3.1
 
-# Newer versions than from the current Zope 4.5.5.  Remove after Zope updates it.
+# Newer versions than from the current Zope 4.6.2.  Remove after Zope updates it.
 
 
 ##############################################################################
@@ -289,7 +289,6 @@ argh = 0.26.2
 args = 0.1.0
 asn1crypto = 1.3.0
 cached-property = 1.5.1
-cffi = 1.14.5
 check-manifest = 0.45
 clint = 0.5.1
 cryptography = 3.3.2
@@ -381,5 +380,3 @@ ply =
     3.4 previously pinned for slimit, but calmjs.parse needs 3.6+, though it comes with pre-generated tab files for up to 3.11. See https://github.com/dabeaz/ply/issues/82,
 Unidecode =
     Unidecode 0.04.{2-9} breaks tests
-zope.interface =
-    5.1 gives several test failures

--- a/versions.cfg
+++ b/versions.cfg
@@ -85,6 +85,11 @@ zope.password = 4.3.1
 
 # Newer versions than from the current Zope 4.6.2.  Remove after Zope updates it.
 
+# Older versions than pinned in Zope.  Remove only if we are sure newer versions are safe.
+# See versionannotations below.
+zope.component = 4.6.2
+# Maybe zope.interface 5.4.0 is too new as well, in that case we should pin:
+# zope.interface = 5.2.0
 
 ##############################################################################
 # External dependencies
@@ -380,3 +385,5 @@ ply =
     3.4 previously pinned for slimit, but calmjs.parse needs 3.6+, though it comes with pre-generated tab files for up to 3.11. See https://github.com/dabeaz/ply/issues/82,
 Unidecode =
     Unidecode 0.04.{2-9} breaks tests
+zope.component =
+    5.0.0 causes a few problems.  See https://github.com/plone/buildout.coredev/pull/725#issuecomment-872272811

